### PR TITLE
Python version upgraded from 3.8 to 3.12

### LINKED ISSUE #3949: Python 3.8 out of support from October 2024, move python version to newer version
## Description

Python 3.8 is moving out of extended support from October 2024. 
References to Python 3.8 need updating and testing.

`Your apps that are hosted on App Service will continue to run, but security updates will no longer be available and we'll no longer provide customer service for Python 3.8. Learn more about App Service language support.`

### PR DESCRIPTION
**Pull request by gauravmishrack**
# Resolves #3949

## What is being addressed

-python version was upgraded from 3.8 to 3.12 as 3.8 is going to be out of support this change will affect in docker image creation versions.

## How is this addressed
Version in upgraded in files

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -10,8 +10,8 @@
     "context": "..",
     "dockerfile": "Dockerfile",
     "args": {
-      // To ensure that the group ID for the docker group in the container
```
Comment (tim-allen-ck) [inline .devcontainer/devcontainer.json:13]: why has "the" been removed?

**Comment by tamirkamara** [comment]
@gauravmishrack @tim-allen-ck is this likely to be completed soon?

**Comment by tim-allen-ck** [comment]
> @gauravmishrack @tim-allen-ck is this likely to be completed soon?

I hope so, going to try and get it finished this week

File: api_app/models/domain/azuretremodel.py [hunk 2]
```diff
@@ -1,7 +1,7 @@
-from pydantic import BaseConfig, BaseModel
+from pydantic import ConfigDict, BaseModel
 
 
 class AzureTREModel(BaseModel):
-    class Config(BaseConfig):
-        allow_population_by_field_name = True
+    class Config(ConfigDict):
```
Comment (marrobi) [inline api_app/models/domain/azuretremodel.py:5]: Is this a pydantic v2 change no longer needed?

File: api_app/models/domain/resource.py [hunk 3]
```diff
@@ -22,11 +23,11 @@ class ResourceHistoryItem(AzureTREModel):
     """
     id: str = Field(title="Id", description="GUID identifying the resource request")
     resourceId: str = Field(title="Id", description="GUID identifying the resource request")
-    properties: dict = Field({}, title="Resource template parameters", description="Parameters for the deployment")
+    properties: dict = Field(None, title="Resource template parameters", description="Parameters for the deployment")
```
Comment (marrobi) [inline api_app/models/domain/resource.py:26]: Are the field changes a pydantic v2 change no longer needed?

File: core/terraform/outputs.sh
Comment (marrobi) [inline core/terraform/outputs.sh:1]: What are these changes for?

**Review by marrobi — COMMENTED** [review]
Just some comments on progress after last weeks conversation.

**Comment by tim-allen-ck** [comment]
Ready for review, I've tested but could do with another person to pull and deploy

**Comment by marrobi** [comment]
> Ready for review, I've tested but could do with another person to pull and deploy

Thank you! Will try give it a test. So most of the issues were due to Pydantic being updated as a dependency?

**Review by Danny-Cooke-CK — APPROVED** [review]
looks good tim

**Comment by tim-allen-ck** [comment]
new PR created https://github.com/microsoft/AzureTRE/pull/4164

**Pull request closed without merging**